src/openms/include/OpenMS/KERNEL/MSExperiment.h

```diff
@@ -697,6 +698,22 @@ namespace OpenMS
       return *this;
     }
 
+    /// get the file path to the first MS run
+    StringList getPrimaryMSRunPath() const
+    {
+      StringList ms_run_paths;
+      std::vector<SourceFile> sfs(this->getSourceFiles());
+      for (std::vector<SourceFile>::const_iterator it = sfs.begin(); it != sfs.end(); ++it)
+      {
+        String ms_run_location = it->getPathToFile() + "/" + it->getNameOfFile();
```

Comment:
is this a smart idea to just use "/" everywhere? does this also work on Windows or will we then have mixed path ?

It seems that we simply copy whatever is there in the `location` attribute of `sourceFile`:

```
# MzMLHandler.h
1248       else if (tag == "sourceFile")
1249       {
1250         current_id_ = attributeAsString_(attributes, s_id);
1251         source_files_[current_id_].setNameOfFile(attributeAsString_(attributes, s_name));
1252         source_files_[current_id_].setPathToFile(attributeAsString_(attributes, s_location));
1253       }
```

According to http://www.peptideatlas.org/tmp/mzML1.1.0.html#sourceFile this could be a windows path, linux path or URI. So maybe this needs some more parsing before we can simply assume that the location of the ms run is path + "/" + name